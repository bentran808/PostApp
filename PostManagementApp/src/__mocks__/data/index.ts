const initialUser = {
  id: '',
  avatar: '',
  name: '',
  gender: '',
  email: '',
  phone: '',
  address: '',
  role: ''
};

const mockAdmin = {
  id: '621c4188dece66a40dad2951',
  avatar: 'https://picsum.photos/200',
  name: 'Admin',
  gender: 'male',
  email: 'admin@admin.com',
  phone: '+84 (919) 507231',
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  role: 'admin'
};

const mockUser = {
  id: '621c41883112f1c216433dcf',
  avatar: 'https://picsum.photos/200',
  name: 'Mcintyre Morrison',
  gender: 'male',
  email: 'test1@email.com',
  phone: '+84 (838) 539391',
  address: '313 Menahan Street, Aurora, North Dakota, 9139',
  role: 'user'
};

const mockPost = {
  author: mockAdmin,
  company: 'bb',
  year: 2222,
  type: 'b',
  status: true,
  price: 23333111,
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  title: 'bb update',
  description: 'bb',
  photos: [
    {
      height: 480,
      uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      width: 480,
      fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      type: 'image/jpeg',
      fileSize: 6689
    },
    {
      height: 480,
      uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      width: 480,
      fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      type: 'image/jpeg',
      fileSize: 6689
    }
  ],
  pending: false,
  createdAt: 1647507076963,
  updatedAt: 1647511146749,
  id: 2
};

const mockLikes = [
  {
    author: mockAdmin,
    postId: 2,
    createdAt: 1647509052801,
    updatedAt: 1647509052801,
    id: 1
  }
];

const mockCommentsList = [
  {
    author: mockAdmin,
    postId: 2,
    content: 'test',
    createdAt: 1647509052801,
    updatedAt: 1647509052801,
    id: 1
  },
  {
    author: {
      id: '621c4188e8365653f9326450',
      avatar: 'https://picsum.photos/200',
      name: 'Frost Spence',
      gender: 'male',
      email: 'frostspence@entogrok.com',
      phone: '+84 (859) 508289',
      address: '745 Blake Avenue, Belmont, District Of Columbia, 4099',
      role: 'user'
    },
    postId: 2,
    content: 'test2',
    createdAt: 1647509052801,
    updatedAt: 0,
    id: 2
  }
];

const mockNewPost = {
  author: mockAdmin,
  company: 'qwe',
  year: 2021,
  type: 'qwe',
  status: true,
  price: 10000,
  address: 'qwe',
  title: 'qwe',
  description: 'qwe',
  photos: [],
  pending: true
};

const mockEditedPost = {
  '2': mockPost,
  '3': {
    id: 3,
    ...mockNewPost
  }
};

const mockApprovedPost = {
  '2': mockPost,
  '3': {
    id: 3,
    ...mockNewPost,
    pending: false
  }
};

const mockLike = {
  author: mockAdmin,
  postId: 2,
  createdAt: 1647509052801,
  updatedAt: 1647509052801,
  id: 2
};

const mockNewComment = {
  author: mockAdmin,
  postId: 2,
  content: 'abc',
  createdAt: 1647509052801,
  updatedAt: 1647509052801,
  id: 3
};

const mockPhotos = [
  {
    fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    fileSize: 6689,
    height: 480,
    type: 'image/jpeg',
    uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    width: 480
  },
  {
    fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    fileSize: 6689,
    height: 480,
    type: 'image/jpeg',
    uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    width: 480
  }
];

export {
  initialUser,
  mockAdmin,
  mockUser,
  mockPost,
  mockLikes,
  mockCommentsList,
  mockNewPost,
  mockEditedPost,
  mockApprovedPost,
  mockLike,
  mockNewComment,
  mockPhotos
};
