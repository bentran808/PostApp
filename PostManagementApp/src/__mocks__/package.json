{
    "name": "__mock__",
    "private": true
}