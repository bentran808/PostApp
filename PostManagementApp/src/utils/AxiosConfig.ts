import axios from 'axios';

export const axiosInstance = axios.create({
    baseURL: 'http://192.168.56.1:3000/',
    headers: {
        'Content-Type': 'application/json'
    }
});
