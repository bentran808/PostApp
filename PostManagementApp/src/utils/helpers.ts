import AsyncStorage from '@react-native-async-storage/async-storage';

const sortDesc = (arr: Post[] = []) => {
  return arr.sort((a, b) => ((a.id || 0) < (b.id || 0) ? 1 : -1));
};

const formatPrice = (price: Number) => {
  return price.toLocaleString('it-IT', {
    style: 'currency',
    currency: 'VND'
  });
};

const getApprovedPosts = (arr: Post[] = []) => arr.filter((item) => !item.pending);

const setAccessToken = async (access_token: string) =>
  await AsyncStorage.setItem('access_token', access_token);

const getAccessToken = async () => await AsyncStorage.getItem('access_token');

const removeAccessToken = async () => await AsyncStorage.removeItem('access_token');

export {
  sortDesc,
  formatPrice,
  getApprovedPosts,
  setAccessToken,
  getAccessToken,
  removeAccessToken
};
