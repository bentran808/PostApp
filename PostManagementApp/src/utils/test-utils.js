import React from 'react';
import { Provider } from 'react-redux';
import { Provider as PaperProvider } from 'react-native-paper';
import { render } from '@testing-library/react-native';
import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import authReducer from 'redux/slices/authSlice';
import postReducer from 'redux/slices/postSlices';

const sagaMiddleware = createSagaMiddleware();
const AllTheProviders = ({ children }) => {
  const store = configureStore({
    reducer: {
      auth: authReducer,
      post: postReducer
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false
      }).concat(sagaMiddleware)
  });
  return (
    <Provider store={store}>
      <PaperProvider>{children}</PaperProvider>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
