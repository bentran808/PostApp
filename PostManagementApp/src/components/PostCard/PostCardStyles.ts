import { StyleSheet } from 'react-native';
import { Colors } from 'react-native-paper';
import { colors } from '../../constants/colors';
import { windowHeight, windowWidth } from '../../utils/Dimensions';

export const styles = StyleSheet.create({
    card: {
        backgroundColor: '#f8f8f8',
        borderRadius: 10,
        marginBottom: 20,
        paddingBottom: 10,
        width: windowWidth * 0.9
    },
    userInfo: {
        flexDirection: 'row',
        justifyContent: 'flex-start',
        padding: 15
    },
    avatar: {
        borderRadius: 25,
        height: 50,
        width: 50
    },
    userInfoText: {
        flexDirection: 'column',
        justifyContent: 'center',
        marginLeft: 10
    },
    userName: {
        fontSize: 14,
        fontWeight: 'bold'
    },
    postTime: {
        fontSize: 12,
        color: '#666',
        marginVertical: 5,
        paddingHorizontal: 15
    },
    postTitle: {
        color: Colors.black,
        fontSize: 14,
        fontWeight: 'bold',
        marginBottom: 5,
        paddingLeft: 15
    },
    postPrice: {
        color: colors.freeSpeechRed,
        fontSize: 14,
        fontWeight: 'bold',
        paddingLeft: 15
    },
    postLocation: {
        flexDirection: 'row',
        paddingLeft: 15
    },
    postAddress: {
        color: Colors.black,
        fontSize: 14,
        marginLeft: 5
    },
    infoSection: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingTop: 10
    },
    postText: {
        fontSize: 14,
        marginBottom: 15,
        paddingHorizontal: 15
    },
    divider: {
        alignSelf: 'center',
        borderBottomColor: colors.gainsboro,
        borderBottomWidth: 1,
        marginTop: 15,
        width: '92%'
    },
    buttonControls: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        paddingVertical: 10
    },
    emptyList: {
        textAlign: 'center',
        marginTop: 10
    },
    interactionWrapper: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        paddingHorizontal: 15,
        paddingTop: 15
    },
    interaction: {
        borderRadius: 5,
        flexDirection: 'row',
        justifyContent: 'center',
        paddingHorizontal: 5,
        paddingVertical: 2
    },
    interactionText: {
        fontSize: 12,
        fontWeight: 'bold',
        marginTop: 5,
        marginLeft: 5
    },
    commentWrapper: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingHorizontal: 15
    },
    commentInput: {
        width: '80%',
        height: windowHeight / 20
    },
    text_sm: {
        fontSize: 12
    }
});
