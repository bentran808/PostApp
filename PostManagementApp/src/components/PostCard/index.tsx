import moment from 'moment';
import React, { useState } from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import Slider from 'react-native-hook-image-slider';
import { Button, Colors, Menu, TextInput } from 'react-native-paper';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { colors } from '../../theme/Colors';
import { useAppSelector } from '../../hooks';
import { selectCurrentUser } from '../../redux/slices';
import { DefaultAvatar } from '../../theme/Images';
import { windowWidth } from '../../utils/Dimensions';
import { formatPrice } from '../../utils/helpers';
import CommentsList from '../CommentsList';
import { styles } from './PostCardStyles';

type PostCardProps = {
  item: Post;
  isMyPost?: Boolean;
  likesOfItem?: PostLike[];
  commentsOfItem?: PostComment[];
  onShowImage: (photos: Photo[]) => void;
  onEdit: () => void;
  onDelete: () => void;
  onLikePost?: (postId: number) => void;
  onUnlikePost?: (likeId: number) => void;
  onAddNewComment?: (postId: number, content: string) => void;
  onDeleteComment?: (commentId: number) => void;
  onEditComment?: (commentId: number, editContent: string) => void;
  onApprovePost?: () => void;
  onRejectPost?: () => void;
};

const PostCard = ({
  item,
  isMyPost = false,
  likesOfItem = [],
  commentsOfItem = [],
  onShowImage,
  onEdit,
  onDelete,
  onLikePost = () => {},
  onUnlikePost = () => {},
  onAddNewComment = () => {},
  onDeleteComment = () => {},
  onEditComment = () => {},
  onApprovePost = () => {},
  onRejectPost = () => {}
}: PostCardProps) => {
  const [visible, setVisible] = useState('');
  const [visibleDetails, setVisibleDetails] = useState(false);
  const [visibleComment, setVisibleComment] = useState(false);
  const [editComment, setEditComment] = useState('');
  const [content, setContent] = useState('');
  const [editContent, setEditContent] = useState('');
  const currentUser = useAppSelector((state) => selectCurrentUser(state));
  const userId = currentUser?.id;
  const isAdmin = currentUser?.role === 'admin';
  const isAuthor = userId === item.author.id || isAdmin;
  const likedPost = likesOfItem.find((like) => like.author.id === userId);
  const numberOfLikes = likesOfItem.length;
  const numberOfComments = commentsOfItem.length;
  const likeIcon = likedPost ? 'heart' : 'heart-outline';
  const likeIconColor = likedPost ? colors.royalBlue : colors.nightRider;
  const likeBackground = likedPost ? '#2e64e515' : 'transparent';
  let likeText = 'Like';
  let commentText = 'Comment';

  if (numberOfLikes === 1) {
    likeText = '1 Like';
  } else if (numberOfLikes > 1) {
    likeText = numberOfLikes + ' Likes';
  }

  if (numberOfComments === 1) {
    commentText = '1 Comment';
  } else if (numberOfComments > 1) {
    commentText = numberOfComments + ' Comments';
  }

  return (
    <View style={styles.card} key={`Post-${item.id}`}>
      <TouchableOpacity testID="showImage" onPress={() => onShowImage(item.photos)}>
        <View style={{ width: windowWidth * 0.9 }}>
          <Slider
            images={item.photos.map((photo: { uri: string }) => photo.uri)}
            imageHeight={200}
          />
        </View>
      </TouchableOpacity>
      <View style={styles.infoSection}>
        <View>
          <Text style={styles.postTitle}>{item.title}</Text>
          <Text style={styles.postPrice}>{formatPrice(Number(item.price))}</Text>
          <Text style={styles.postTime}>{moment(new Date(item.updatedAt || 0)).fromNow()}</Text>
          <View style={styles.postLocation}>
            <Ionicons name="md-location-outline" size={20} color={Colors.black} />
            <Text style={styles.postAddress}>{item.address}</Text>
          </View>
        </View>
        {isAuthor && (
          <Menu
            visible={visible === `post-${item.id}`}
            onDismiss={() => setVisible('')}
            anchor={
              <TouchableOpacity onPress={() => setVisible(`post-${item.id}`)} testID="Menu">
                <Ionicons name="ellipsis-vertical" size={20} color={Colors.black} />
              </TouchableOpacity>
            }
          >
            <Menu.Item
              onPress={() => {
                setVisible('');
                onEdit();
              }}
              title="Edit"
              testID="MenuEdit"
            />
            <Menu.Item
              onPress={() => {
                setVisible('');
                onDelete();
              }}
              title="Delete"
              testID="Menu.Delete"
            />
          </Menu>
        )}
      </View>
      {visibleDetails && (
        <View testID="contentDetails">
          <View style={styles.divider} />
          <View style={styles.userInfo}>
            <Image
              defaultSource={DefaultAvatar}
              source={{
                uri: item.author.avatar
              }}
              style={styles.avatar}
            />
            <View style={styles.userInfoText}>
              <Text style={styles.userName}>{item.author.name}</Text>
            </View>
          </View>
          <Text style={styles.postText}>{item.description}</Text>
          <View style={styles.postLocation}>
            <FontAwesome name="building-o" size={20} color={Colors.black} />
            <Text style={styles.postAddress}>Product Company: {item.company}</Text>
          </View>
          <View style={styles.postLocation}>
            <Ionicons name="pricetags-outline" size={20} color={Colors.black} />
            <Text style={styles.postAddress}>Type of product: {item.type}</Text>
          </View>
          <View style={styles.postLocation}>
            <Ionicons name="md-calendar-outline" size={20} color={Colors.black} />
            <Text style={styles.postAddress}>Year of registration: {item.year}</Text>
          </View>
          <View style={styles.postLocation}>
            <Ionicons name="md-card-outline" size={20} color={Colors.black} />
            <Text style={styles.postAddress}>Status: {item.status ? 'Used' : 'New'}</Text>
          </View>
        </View>
      )}
      <Button
        testID="showDetails"
        color={colors.nightRider}
        icon={visibleDetails ? 'chevron-up' : 'chevron-down'}
        labelStyle={styles.text_sm}
        uppercase={false}
        onPress={() => setVisibleDetails(!visibleDetails)}
      >
        {visibleDetails ? 'Show less' : 'Show more'}
      </Button>
      {isMyPost ? (
        isAdmin ? (
          <>
            <View style={styles.divider} />
            <View style={styles.buttonControls}>
              <Button
                mode="contained"
                icon="check"
                color={colors.royalBlue}
                onPress={onApprovePost}
              >
                Approve
              </Button>
              <Button
                mode="contained"
                icon="close"
                color={colors.freeSpeechRed}
                onPress={onRejectPost}
              >
                Reject
              </Button>
            </View>
          </>
        ) : (
          item.pending && (
            <>
              <View style={styles.divider} />
              <Text style={styles.emptyList}>This post is pending approval</Text>
            </>
          )
        )
      ) : (
        <>
          <View style={styles.divider} />
          <View style={styles.interactionWrapper}>
            <TouchableOpacity
              onPress={() => (likedPost ? onUnlikePost(likedPost.id) : onLikePost(item.id || 0))}
              style={[
                styles.interaction,
                {
                  backgroundColor: likeBackground
                }
              ]}
              testID="likeButton"
            >
              <Ionicons name={likeIcon} size={25} color={likeIconColor} />
              <Text style={[styles.interactionText, { color: likeIconColor }]}>{likeText}</Text>
            </TouchableOpacity>
            <TouchableOpacity
              style={styles.interaction}
              onPress={() => {
                setVisibleComment(true);
              }}
              testID="commentButton"
            >
              <Ionicons name="md-chatbubble-outline" size={25} />
              <Text style={styles.interactionText}>{commentText}</Text>
            </TouchableOpacity>
          </View>
        </>
      )}
      {visibleComment && (
        <>
          {numberOfComments ? <View style={styles.divider} /> : undefined}
          <CommentsList
            commentsOfItem={commentsOfItem}
            visible={visible}
            editComment={editComment}
            editContent={editContent}
            setVisible={setVisible}
            setEditComment={setEditComment}
            onSetEditContent={setEditContent}
            onDeleteComment={onDeleteComment}
            onEditComment={onEditComment}
          />
          <View style={styles.divider} />
          <View style={styles.commentWrapper} testID="commentWrapper">
            <TextInput
              mode="outlined"
              value={content}
              activeOutlineColor={colors.royalBlue}
              style={styles.commentInput}
              onChangeText={(itemValue) => setContent(itemValue)}
              placeholder="Write a comment"
              right={
                content ? (
                  <TextInput.Icon
                    name="close-circle-outline"
                    size={18}
                    onPress={() => setContent('')}
                  />
                ) : undefined
              }
              testID="inputComment"
            />
            <Button
              color={colors.royalBlue}
              uppercase={false}
              disabled={content === ''}
              onPress={() => {
                onAddNewComment(item.id || 0, content);
                setContent('');
              }}
              testID="buttonAddComment"
            >
              Send
            </Button>
          </View>
        </>
      )}
    </View>
  );
};

export default PostCard;
