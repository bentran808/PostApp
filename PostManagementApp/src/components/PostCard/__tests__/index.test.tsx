import React from 'react';
import * as customHooks from '../../../hooks';
import { fireEvent, render, RenderAPI } from '../../../utils/test-utils';
import { mockCommentsList, mockLikes, mockPost, mockUser } from '../../../__mocks__/data';
import PostCard from '../index';

describe('Post Card Component', () => {
  const spy = jest.spyOn(customHooks, 'useAppSelector');
  spy.mockReturnValue(mockUser);

  let props: JSX.IntrinsicAttributes & {
    item: Post;
    isMyPost?: Boolean | undefined;
    likesOfItem?: PostLike[] | undefined;
    commentsOfItem?: PostComment[] | undefined;
    onShowImage: (photos: Photo[]) => void;
    onEdit: () => void;
    onDelete: () => void;
    onLikePost?: ((postId: number) => void) | undefined;
    onUnlikePost?: ((likeId: number) => void) | undefined;
    onAddNewComment?: ((postId: number, content: string) => void) | undefined;
    onDeleteComment?: ((commentId: number) => void) | undefined;
    onEditComment?: ((commentId: number, editContent: string) => void) | undefined;
    onApprovePost?: (() => void) | undefined;
    onRejectPost?: (() => void) | undefined;
  };
  let component: RenderAPI;

  beforeAll(() => {
    props = {
      item: mockPost,
      likesOfItem: mockLikes,
      commentsOfItem: mockCommentsList,
      onShowImage: jest.fn(),
      onEdit: jest.fn(),
      onDelete: jest.fn(),
      onLikePost: jest.fn(),
      onUnlikePost: jest.fn()
    };

    component = render(<PostCard {...props} />);
  });

  test('renders correctly', () => {
    const menu = component.getAllByTestId('Menu');
    fireEvent.press(menu[0]);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should show image with data', () => {
    const { getByTestId } = render(<PostCard {...props} />);
    const button = getByTestId('showImage');
    fireEvent.press(button);
    expect(props.onShowImage).toBeCalledWith([
      {
        fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
        fileSize: 6689,
        height: 480,
        type: 'image/jpeg',
        uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
        width: 480
      }
    ]);
  });

  test('should details post', () => {
    const { getByTestId } = render(<PostCard {...props} />);
    const button = getByTestId('showDetails');
    fireEvent.press(button);
    expect(getByTestId('contentDetails')).toBeDefined();
  });

  test('should call like button', () => {
    const { getByTestId } = render(<PostCard {...props} />);
    const button = getByTestId('likeButton');
    fireEvent.press(button);
    expect(props.onUnlikePost).toBeCalledWith(1);
  });

  test('should call comment button', () => {
    const { getByTestId } = render(<PostCard {...props} />);
    const button = getByTestId('commentButton');
    fireEvent.press(button);

    const input = getByTestId('inputComment');
    fireEvent.changeText(input, 'test');

    const buttonSend = getByTestId('buttonAddComment');
    fireEvent.press(buttonSend);

    expect(getByTestId('commentWrapper')).toBeDefined();
    expect(getByTestId('commentList')).not.toBeNull();
  });
});
