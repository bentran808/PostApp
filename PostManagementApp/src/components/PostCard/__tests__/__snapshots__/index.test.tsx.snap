// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Card Component renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f8f8f8",
        "borderRadius": 10,
        "marginBottom": 20,
        "paddingBottom": 10,
        "width": 675,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="showImage"
    >
      <View
        style={
          Object {
            "width": 675,
          }
        }
      >
        <View
          style={
            Object {
              "height": 200,
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <RCTScrollView
            bounces={false}
            data={
              Array [
                "file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg",
              ]
            }
            decelerationRate="fast"
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onViewableItemsChanged={[Function]}
            pagingEnabled={true}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "height": "100%",
                "width": 750,
              }
            }
            viewabilityConfig={
              Object {
                "viewAreaCoveragePercentThreshold": 60,
                "waitForInteraction": false,
              }
            }
            viewabilityConfigCallbackPairs={
              Array [
                Object {
                  "onViewableItemsChanged": [Function],
                  "viewabilityConfig": Object {
                    "viewAreaCoveragePercentThreshold": 60,
                    "waitForInteraction": false,
                  },
                },
              ]
            }
          >
            <View>
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                    null,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#FFF",
                      "height": "auto",
                      "width": 750,
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg",
                      }
                    }
                    style={
                      Object {
                        "height": "100%",
                        "overflow": "hidden",
                        "resizeMode": "cover",
                        "width": 750,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#D3D3D3",
                        "height": "100%",
                        "justifyContent": "center",
                        "position": "absolute",
                        "width": 750,
                      }
                    }
                  >
                    <ActivityIndicator
                      color="#000"
                      size="large"
                    />
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 25,
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFF",
                    "borderColor": "#FFF",
                    "borderRadius": 50,
                    "borderWidth": 2,
                    "height": 20,
                    "marginRight": 7,
                    "width": 20,
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 10,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
              "paddingLeft": 15,
            }
          }
        >
          bb update
        </Text>
        <Text
          style={
            Object {
              "color": "#b80d0d",
              "fontSize": 14,
              "fontWeight": "bold",
              "paddingLeft": 15,
            }
          }
        >
          23.333.111 VND
        </Text>
        <Text
          style={
            Object {
              "color": "#666",
              "fontSize": 12,
              "marginVertical": 5,
              "paddingHorizontal": 15,
            }
          }
        >
          5 days ago
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingLeft": 15,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 14,
                "marginLeft": 5,
              }
            }
          >
            199 Bergen Street, Bowie, Oregon, 7933
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="Menu"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="showDetails"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "lineHeight": 12,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰅀
            </Text>
          </View>
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  false,
                  Object {
                    "color": "#333",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            Show more
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderBottomColor": "#ddd",
          "borderBottomWidth": 1,
          "marginTop": 15,
          "width": "92%",
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "paddingHorizontal": 15,
          "paddingTop": 15,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2e64e515",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 2,
          }
        }
        testID="likeButton"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#2e64e5",
                "fontSize": 25,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginLeft": 5,
                "marginTop": 5,
              },
              Object {
                "color": "#2e64e5",
              },
            ]
          }
        >
          1 Like
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 2,
          }
        }
        testID="commentButton"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 25,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginLeft": 5,
              "marginTop": 5,
            }
          }
        >
          2 Comments
        </Text>
      </View>
    </View>
  </View>
</View>
`;
